aea/agent.py

```diff
@@ -54,12 +54,70 @@ def is_stopped(self) -> bool:
         return self._is_stopped
 
 
+class Identity:
+    """
+    An identity are the public elements identifying an agent.
+
+    It can include:
+    - the agent name
+    - the public half of private / public key pairs
+    - the addresses
+    """
+
+    def __init__(self, name: str):
```

Comment:
I think the constructor should include public keys and addresses. It seems more natural, as the `Identity` is a data holder, and its content should be immutable, e.g. read-only. 

You can optionally have empty dictionaries as defaults, both for public keys and addresses. They can then be populated manually (by accessing the properties `public_keys` and `addresses`), or by invoking ad-hoc methods (e.g. `add_address`), that make some consistency check before adding/removing items, potentially raising `ValueError`.

Reply:
How about we just add `addresses` and drop `public_keys`? Public keys aren't needed by the agents.

Reply:
yes